### PR TITLE
Switch tasks to pytest

### PR DESCRIPTION
**Pull request by olegsinyavskiy**
@fchollet , @phreeza please take a look.  While github diff looks terrible, I didn't do much, just few renames and fixes. Here I just want to throw few ideas in to avoid spending time in the future.

Small cleanup of test_tasks:
 - switch to pytest
 - fix random seed to be independent on the order of imports
 - readable test names
 - remove unnecessary prints

Experimental:
 - switch some tests to use list of layers in constructor. I was always wandering why torch do not do that - just pass a list of layers into Sequential. I was pleasantly surprised to find this in Keras.

### CODE REVIEW

**Comment by fchollet** [comment]
> switch some tests to use list of layers in constructor.

Could you clarify what you mean?

I think these are reasonable changes. Regarding prints, we should keep prints that make the test logs easier to read, specifically prints such as "now testing this and that...". 

Also: any assert should have a readable failure message associated to it. If a test fails because some value was too high or something, we want to know what the value was (and what it was being compared to).

File: .gitignore [hunk 1]
```diff
@@ -9,3 +9,10 @@ keras/datasets/temp/*
 docs/site/*
 docs/theme/*
 tags
+
+# test-related
+.coverage
+.cache
+
+# developer environments
+.idea
```
Comment (fchollet) [inline .gitignore]: What's this?

Comment (olegsinyavskiy) [inline .gitignore]: .coverage
.cache
coverage files that you get in you working tree if you run tests locally

.idea is pycharm

**Comment by olegsinyavskiy** [comment]
@fchollet, this is exactly what you get with pytest - I renamed functions to be the same as prints so you get the same as you had. Also pytest will generate you a nice assert message if it fails

**Comment by fchollet** [comment]
Ok, sounds good.

**Pull request merged**